src/modules/Elsa.Workflows.Core/Contexts/WorkflowExecutionContext.cs

```diff
@@ -670,7 +530,7 @@ subStatus switch
             _ => throw new ArgumentOutOfRangeException(nameof(subStatus), subStatus, null)
         };
 
-    private bool ValidateStatusTransition(WorkflowSubStatus targetSubStatus)
+    private bool ValidateStatusTransition()
     {
         var currentMainStatus = GetMainStatus(SubStatus);
```

Comment:
I think this is the wrong fix and instead we should change this line, to:

```csharp
if (!ValidateStatusTransition(subStatus))
```

Reply:
I did not make a fix here, just removed code that was not used. The validate method is not so much checking if the new state can be moved to as it is checking if the current state is finished and therefor a state that can be moved from. Sending the target sub status and using it instead of the current `SubStatus` will prevent us from moving to the Finished state.

Reply:
I understand. I'm just wondering if the original code is a mistake and whether it should have been using the parameter instead of the property. I'll add a TODO to follow up.